arbnode/mel/extraction/message_extraction_function_test.go

```diff
@@ -0,0 +1,378 @@
+package melextraction
+
+import (
+	"context"
+	"errors"
+	"io"
+	"math/big"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/ethereum/go-ethereum/trie"
+
+	"github.com/offchainlabs/nitro/arbnode/mel"
+	"github.com/offchainlabs/nitro/arbos/arbostypes"
+	"github.com/offchainlabs/nitro/arbstate"
+	"github.com/offchainlabs/nitro/daprovider"
+)
+
+func TestExtractMessages(t *testing.T) {
```

Comment:
I don't love the structure of this test because it's hard to follow for each test case, exactly the test conditions that are being checked. Maybe if you moved most of the helper functions outside of the test function it would be a little clearer?

Like, they aren't closing over any state internal to the test function, are they?